src/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -87,6 +87,43 @@ public static void ReadSimpleClassWithObjectArray()
             obj.Verify();
         }
 
+        [Fact]
+        public static void ReadArrayInObjectArray()
+        {
+            object[] array = JsonSerializer.Parse<object[]>(@"[[]]");
+            Assert.Equal(1, array.Length);
+            Assert.IsType<JsonElement>(array[0]);
+            Assert.Equal(JsonValueType.Array, ((JsonElement)array[0]).Type);
+        }
+
+        [Fact]
+        public static void ReadObjectInObjectArray()
+        {
+            object[] array = JsonSerializer.Parse<object[]>(@"[{}]");
```

Comment:
Let's add a couple more tests for completeness:

Array with nested array:
```text
[1, false, { ""name"" : ""Person"" }, [1, 2, 3]]
```

Array with object, with another nesting:
```text
[1, false, { ""name"" : [1, false, { ""name"" : ""Person"" }] }]
```